src/Mono.Android/Java.Interop/TypeManager.cs

```diff
@@ -318,7 +318,7 @@ public static void RegisterType (string java_class, Type t)
 					if (jniFromType != java_class) {
 						managedToJni.Add (t, java_class);
 					}
-				} else {
+				} else if (!ignoreExistingRegistration) {
```

Comment:
From our point of view it's the same reason gref logging is not enabled by default for instance.